jina/types/score/map.py

```diff
@@ -0,0 +1,102 @@
+from typing import Optional, Union
+
+import numpy as np
+
+from ..score import NamedScore
+from ..mixin import ProtoTypeMixin
+from ...excepts import BadNamedScoreType
+from ...helper import typename
+from ...proto import jina_pb2
+
+__all__ = ['MappedNamedScore']
+
+
+class MappedNamedScore(ProtoTypeMixin):
+
+    """
+    :class:`MappedNamedScore` is one of the **primitive data type** in Jina.
+
+    It offers a Pythonic interface to allow users access and manipulate
+    :class:`jina.jina_pb2.MappedNamedScoreProto` object without working with Protobuf itself.
+
+    It offers an interface to access and update scores as `NamedScore` as values of a `dict` with a string key.
+
+    To create a :class:`MappedNamedScoreProto` object, simply:
+
+        .. highlight:: python
+        .. code-block:: python
+
+            from jina.types.score.map import MappedNamedScore
+            scores = MappedNamedScore()
+            scores['score'] = 50
+
+    :class:`MappedNamedScore` can be built from ``jina_pb2.MappedNamedScoreProto`` (as a weak reference or a deep copy)
+
+    :param scores: The scores to construct from, depending on the ``copy``,
+        it builds a view or a copy from it.
+    :type score: Optional[jina_pb2.MappedNamedScoreProto]
+    :param copy: When ``scores`` is given as a :class:`MappedNamedScoreProto` object, build a
+        view (i.e. weak reference) from it or a deep copy from it.
+    :type copy: bool
+    :param kwargs: Other parameters to be set
+
+    """
+
+    def __init__(
+        self,
+        scores: Optional[jina_pb2.MappedNamedScoreProto] = None,
+        copy: bool = False,
+        **kwargs,
+    ):
+        self._pb_body = jina_pb2.MappedNamedScoreProto()
+        try:
+            if isinstance(scores, jina_pb2.MappedNamedScoreProto):
+                if copy:
+                    self._pb_body.CopyFrom(scores)
+                else:
+                    self._pb_body = scores
+            elif scores is not None:
+                # note ``None`` is not considered as a bad type
+                raise ValueError(f'{typename(scores)} is not recognizable')
+        except Exception as ex:
+            raise BadNamedScoreType(
+                f'fail to construct a MappedNamedScore from {scores}'
+            ) from ex
+
+    def __setitem__(
+        self,
+        key: str,
+        value: Union[jina_pb2.NamedScoreProto, NamedScore, float, np.generic],
```

Comment:
i think it's safe to remove `NamedScore` unless we really want to keep the fields other than `op_name` and `value`

Reply:
so far we want to keep them

Reply:
okay